lasagne/tests/layers/test_dense.py

```diff
@@ -113,3 +115,173 @@ def test_named_layer_param_names(self, DenseLayer, dummy_input_layer):
 
         assert layer.W.name == "foo.W"
         assert layer.b.name == "foo.b"
+
+
+class TestNonlinearityLayer:
+    @pytest.fixture
+    def NonlinearityLayer(self):
+        from lasagne.layers.dense import NonlinearityLayer
+        return NonlinearityLayer
+
+    @pytest.fixture
+    def layer_vars(self, dummy_input_layer):
+        from lasagne.layers.dense import NonlinearityLayer
+        nonlinearity = Mock()
+
+        layer = NonlinearityLayer(
+            dummy_input_layer,
+            nonlinearity=nonlinearity,
+            )
+
+        return {
+            'nonlinearity': nonlinearity,
+            'layer': layer,
+            }
+
+    @pytest.fixture
+    def layer(self, layer_vars):
+        return layer_vars['layer']
+
+    def test_init_none_nonlinearity(self, NonlinearityLayer,
+                                    dummy_input_layer):
+        layer = NonlinearityLayer(
+            dummy_input_layer,
+            nonlinearity=None,
+            )
+        assert layer.nonlinearity == lasagne.nonlinearities.identity
+
+    def test_get_output_for(self, layer_vars):
+        layer = layer_vars['layer']
+        nonlinearity = layer_vars['nonlinearity']
+
+        input = theano.shared(np.random.uniform(-1, 1, (2, 12)))
+        result = layer.get_output_for(input)
+        assert result is nonlinearity.return_value
```

Comment:
`input = theano.tensor.matrix()` would be enough, and an `assert nonlinearity.called_with(input)` would be a nice addition!